src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java

```diff
@@ -319,7 +329,8 @@ public BukkitTask runTaskLater(Plugin plugin, BukkitRunnable task, long delay)
 	public BukkitTask runTaskLaterAsynchronously(Plugin plugin, Runnable task, long delay)
 	{
 		ScheduledTask scheduledTask = new ScheduledTask(id++, plugin, false, currentTick + delay,
-		        new AsyncRunnable(task));
+				new AsyncRunnable(task));
+		scheduledTask.addOnCancelled(() -> asyncTasksQueued--);
```

Comment:
This is an asynchronous task, it is likely to be canceled in a different thread. Shouldn't you use an atomic long for this kind of actions instead?

Reply:
This should indeed probably be an atomic